goth_tests/README.md

```diff
@@ -0,0 +1,267 @@
+```
+providers = runner.get_probes(probe_type=ProviderProbe)
+requestor = runner.get_probes(probe_type=RequestorProbe)[0]
+```
+
+`Probe` is `goth`'s interface for interacting with `yagna` nodes running as part of a test. These objects contain modules for interacting with the node's REST API, executing CLI commands and interacting with the Docker API.
+
+Finally, the sample test includes some simple examples of how these modules can be used:
+```
+drivers = requestor.cli.payment_drivers()
+assert drivers and drivers.items()
+
+for provider in providers:
+    await provider.wait_for_offer_subscribed()
+```
+
+There's much more to `goth` that's not covered in this overview. Some of the more advanced features are: event assertions, event monitors, running host commands etc.
+
+Make sure to take a look at the integration tests in this repository, as well as the [ones in `yapapi`](https://github.com/golemfactory/yapapi/tree/master/tests/goth).
+
+### Test configuration
+
+#### `goth-config.yml`
+`goth` can be configured using a YAML file. The default `goth-config.yml` is located in the common, generated assets (i.e. `goth_tests/assets`) and looks something like this:
+```
+docker-compose:
+
+  docker-dir: "docker"                          # Where to look for docker-compose.yml and Dockerfiles
+
+  build-environment:                            # Fields related to building the yagna Docker image
+    # binary-path: ...
+    # deb-path: ...
+    # branch: ...
+    # commit-hash: ...
+    # release-tag: ...
+
+  compose-log-patterns:                         # Log message patterns used for container ready checks
+    ethereum: ".*Wallets supplied."
+    ...
+
+key-dir: "keys"                                 # Where to look for pre-funded Ethereum keys
+
+node-types:                                     # User-defined node types to be used in `nodes`
+  - name: "Requestor"
+    class: "goth.runner.probe.RequestorProbe"
+
+  - name: "Provider"
+    class: "goth.runner.probe.ProviderProbe"
+    mount: ...
+
+nodes:                                          # List of yagna nodes to be run in the test
+  - name: "requestor"
+    type: "Requestor"
+
+  - name: "provider-1"
+    type: "Provider"
+    use-proxy: True
+```
+
+#### Configuration overrides
+It's also possible to override parts of the configuration. This is useful for whenever you want to change some config value without having to edit the `goth-config.yml` file.
+
+When running tests you can use the `--config-override` parameter:
+```
+poetry run poe goth-tests --config-override docker-compose.build-environment.commit-hash=29b7f85
+```
+
+Overrides specified as CLI arguments are passed into the test functions through the `config_overrides` fixture:
+```
+async def test_something(
+    config_overrides: Optional[List[Override]],     # This is provided by a fixture
+    ...
+):
+    goth_config = load_yaml(Path("path/to/goth-config.yml", config_overrides)
+```
```

Comment:
This `README` is excellent!

Reply:
Indeed. Much appreciated effort @zakaprov

Reply:
❤️